.NET Core - unit test runner returning exit code 3

Build 23 failed:

> Tests passed: 266, ignored: 1; exit code 3 (new) 	
http://builds.castleproject.org/viewType.html?buildTypeId=CoreNetCore_MasterNetCore

Hmm. With the latest changes from @jeremymeng, we don't get so far as to run the tests. Stuck on 

```
…\src\Castle.Core.Tests\project.json(9,20): error NU1002: The dependency Castle.Core 3.3.4 in project Castle.Core.Tests does not support framework DNXCore,Version=v5.0.
```

which I can reproduce at home. Yay! I'm not sure what to do, though. We appear to build Castle.Core for `dotnet5.4`, but then try to build `Castle.Core.Tests` for framework `dnxcore50`.It looks like 0a55378fd9cc changed Castle.Core to build `dotnet5.4` instead of `dotnet`. Unfortunately, this is all Greek to me. Other than "it's different", I don't understand what the differences mean. Changing Castle.Core.Tests to build against framework `dotnet5.4` just made it angry that we don't have a proper xUnit:

```
…\src\Castle.Core.Tests\project.json(15,25): error NU1002: The dependency xunit.runner.dnx 2.1.0-rc1-build204 in project Castle.Core.Tests does not support framework .NETPlatform,Version=v5.4.
```

Out of my depth and sleepy. Will come back in several hours. Sorry.

--- a/.gitignore
+++ b/.gitignore
@@ -14,3 +14,4 @@ artifacts/
 src/**/obj/
 src/**/bin/
 project.lock.json
+/.dnx/

--- a/buildscripts/Build-NetCore.ps1
+++ b/buildscripts/Build-NetCore.ps1
@@ -1,0 +1,39 @@
+$frameworkVersion = "1.0.0-rc1-update1"
+$frameworkArchitecture = "x86"
+
+$dnxHome = Join-Path (Get-Location) ".dnx"
+$dnxPath = Join-Path $dnxHome "bin"
+
+Write-Host "Downloading dnvm to: $dnxPath"
+if (!(Test-Path $dnxPath)) { md $dnxPath | Out-Null }
+
+$dnvmPs1Path = Join-Path $dnxPath "dnvm.ps1"
+$dnvmCmdPath = Join-Path $dnxPath "dnvm.cmd"
+
+$wc = New-Object System.Net.WebClient
+$wc.DownloadFile('https://raw.githubusercontent.com/aspnet/Home/dev/dnvm.ps1', $dnvmPs1Path)
+$wc.DownloadFile('https://raw.githubusercontent.com/aspnet/Home/dev/dnvm.cmd', $dnvmCmdPath)
+
+Write-Host "Downloading dnx"
+
+$env:PATH = ("$dnxPath;" + $env:PATH)
+
+dnvm install $frameworkVersion -runtime coreclr -arch $frameworkArchitecture
+
+dnx --version
+
+Write-Host "Downloading packages"
+
+dnu restore
+
+Write-Host "Building"
+
+$env:Path = ((Join-Path $dnxHome "runtimes\dnx-coreclr-win-$frameworkArchitecture.$frameworkVersion\bin") + ";" + $env:Path)
+
+dnu build src/Castle.Core src/Castle.Core.Tests --configuration Release --out build/NETCORE
+
+Write-Host "Running tests"
+
+dnx -p src/Castle.Core.Tests test
+
+exit $LASTEXITCODE

--- a/buildscripts/Build-NetCore.ps1
+++ b/buildscripts/Build-NetCore.ps1
@@ -34,6 +34,7 @@ dnu build src/Castle.Core src/Castle.Core.Tests --configuration Release --out bu
 
 Write-Host "Running tests"
 
-dnx -p src/Castle.Core.Tests test
+cd src/Castle.Core.Tests
+dnx test
 
 exit $LASTEXITCODE

--- a/buildscripts/build.cmd
+++ b/buildscripts/build.cmd
@@ -19,6 +19,13 @@ IF NOT EXIST %~dp0..\Settings.proj GOTO msbuild_not_configured
 REM Set Framework version based on passed in parameter
 IF "%1" == "" goto no_nothing
 
+REM nesting the netcore build in an IF body prevents reporting of the error level
+IF /i "%1" NEQ "NETCORE" goto not_netcore
+
+PowerShell.exe -NoProfile -NonInteractive -NoLogo -ExecutionPolicy Unrestricted -Command %~dp0Build-NetCore.ps1
+EXIT /B %ERRORLEVEL%
+
+:not_netcore
 IF /i "%1" == "NET40" (SET FrameworkVersion=v4.0)
 IF /i "%1" == "NET40" (SET BuildConfigKey=NET40)
 

